fix(deno): allow test to succeed if there are no tests to run

charsleysa opened the pull request.

barbados-clemens commented on packages/deno/src/executors/test/test.impl.ts:
```diff
@@ -137,6 +137,8 @@ function createArgs(options: DenoTestExecutorNormalizedSchema) {
     args.push(`--shuffle=${options.shuffle}`);
   }
 
+  args.push('--allow-none');
+
```
This should be configurable.

Can you add a new executor option for this?

barbados-clemens reviewed (changes requested):
We should make this configurable. I'm okay with making it default to be on, like we do with @nx/jest, but people should be able to opt out if they so wish.

charsleysa commented:
@barbados-clemens I've updated the PR according to your feedback.

I have added `allowNone` as an option. I have **not** made it default to `true` inside the executor in order to keep the functionality the same for existing projects. I have added the option to the generators so that new projects are generated with `allowNone` set to `true`.

barbados-clemens commented on packages/deno/src/generators/application/__snapshots__/application.spec.ts.snap:
```diff
@@ -35,6 +35,7 @@ Object {
     "test": Object {
       "executor": "@nx/deno:test",
       "options": Object {
+        "allowNone": true,
```
not needed since the schema is defaulted to true

barbados-clemens commented on packages/deno/src/generators/application/__snapshots__/application.spec.ts.snap:
```diff
@@ -81,6 +82,7 @@ Object {
     "test": Object {
       "executor": "@nx/deno:test",
       "options": Object {
+        "allowNone": true,
```
not needed since the schema is defaulted to true

barbados-clemens commented on packages/deno/src/generators/application/lib.ts:
```diff
@@ -106,6 +106,7 @@ export function addProjectConfig(tree: Tree, opts: DenoAppNormalizedSchema) {
       options: {
         coverageDirectory,
         denoConfig: joinPathFragments(opts.projectRoot, 'deno.json'),
+        allowNone: true,
```
not needed since the schema is defaulted to true

barbados-clemens commented on packages/deno/src/generators/serverless/__snapshots__/serverless.spec.ts.snap:
```diff
@@ -41,6 +41,7 @@ Object {
     "test": Object {
       "executor": "@nx/deno:test",
       "options": Object {
+        "allowNone": true,
```
not needed since the schema is defaulted to true

barbados-clemens commented on packages/deno/src/generators/serverless/serverless.ts:
```diff
@@ -120,6 +120,7 @@ export function addProject(tree: Tree, opts: NormalizedSchema) {
       options: {
         coverageDirectory,
         denoConfig: joinPathFragments(opts.projectRoot, 'deno.json'),
+        allowNone: true,
```
not needed since the schema is defaulted to true

barbados-clemens commented on packages/deno/src/generators/utils/testing/deno-app.ts:
```diff
@@ -32,6 +32,7 @@ export function createDenoAppForTesting(
         options: {
           coverageDirectory: joinPathFragments('coverage', opts.projectRoot),
           denoConfig: joinPathFragments(opts.projectRoot, 'deno.json'),
+          allowNone: true,
```
not needed since the schema is defaulted to true

barbados-clemens reviewed (changes requested):
I am okay with changing this as a default as it's more or less the behavior expected IMO.

You did set the default to be true, so go a head and remove setting in the project configuration and leave it true. 

If you're worried about breaking people then you can add a migration to update that property for everyone to be 'false' so it doesn't change, but IMO it's not a big deal.

charsleysa commented:
@barbados-clemens thanks for the feedback! Looks like I misunderstood how the default field in the schema functioned. I've made the changes as recommended.
